miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/analytics/MiniAppAnalytics.kt

```diff
@@ -17,34 +18,87 @@ private inline fun <T> whenHasAnalytics(callback: () -> T) {
 
 /** Only init when analytics dependency is provided. */
 @Suppress("SwallowedException", "TooGenericExceptionCaught")
-internal class MiniAppAnalytics(val rasProjectId: String) {
+internal class MiniAppAnalytics(
+    val rasProjectId: String,
+    private val configs: List<MiniAppAnalyticsConfig>
+) {
 
     companion object {
-        var instance: MiniAppAnalytics? = null
-
-        fun init(rasProjectId: String) = whenHasAnalytics {
-            instance = MiniAppAnalytics(rasProjectId)
+        @Suppress("LongMethod")
+        internal fun sendAnalyticsDefault(
+            rasProjectId: String,
+            eType: Etype,
+            actype: Actype,
+            miniAppInfo: MiniAppInfo?
+        ) = try {
+            val params = createParams(
+                rasProjectId = rasProjectId,
+                acc = BuildConfig.ANALYTICS_ACC,
+                aid = BuildConfig.ANALYTICS_AID,
+                actype = actype,
+                miniAppInfo = miniAppInfo
+            )
+            whenHasAnalytics {
+                RatTracker.event(eType.value, params).track()
+            }
+        } catch (e: Exception) {
+            Log.e("MiniAppAnalytics", e.message.orEmpty())
         }
-    }
 
-    fun sendAnalytics(eType: Etype, actype: Actype, miniAppInfo: MiniAppInfo?) = try {
-        val params = mutableMapOf<String, Any>()
-        // Send to this acc/aid
-        params["acc"] = BuildConfig.ANALYTICS_ACC
-        params["aid"] = BuildConfig.ANALYTICS_AID
+        /** common function to create params to send to tracker. */
+        @Suppress("LongMethod")
+        @VisibleForTesting
+        internal fun createParams(
+            rasProjectId: String,
+            acc: Int,
+            aid: Int,
+            actype: Actype,
+            miniAppInfo: MiniAppInfo?
+        ): MutableMap<String, Any> {
+            val params = mutableMapOf<String, Any>()
```

Comment:
Similar to my comment above - it's better to use Immutable objects when possible. So maybe you could do something like this:

```
val cp = JSONObject()
    .put("mini_app_project_id", rasProjectId)
    .put("mini_app_sdk_version", BuildConfig.VERSION_NAME)
if (miniAppInfo != null) {
    cp.put("mini_app_id", miniAppInfo.id)
        .put("mini_app_version_id", miniAppInfo.version.versionId)
}

return mapOf<String, Any>(
    "acc" to acc,
    "aid" to aid,
    "actype" to actype.value,
    "cp" to cp
)
```